SpeechTranslator/MainPage.xaml.cs

```diff
@@ -0,0 +1,512 @@
+        {
+            await InitializeRecognizer(en_lang);
+            await Initializesynthesizer();
+        }
+        private async Task InitializeRecognizer(Language recognizerLanguage)
+        {
+            if (speechRecognizer != null)
+            {
+                speechRecognizer.ContinuousRecognitionSession.Completed -= ContinuousRecognitionSession_Completed;
+                speechRecognizer.ContinuousRecognitionSession.ResultGenerated -= ContinuousRecognitionSession_ResultGenerated;
+                speechRecognizer.HypothesisGenerated -= SpeechRecognizer_HypothesisGenerated;
+                speechRecognizer.Dispose();
+                speechRecognizer = null;
+
+            }
+            speechRecognizer = new SpeechRecognizer(recognizerLanguage);
+            var dictationConstraint = new SpeechRecognitionTopicConstraint(SpeechRecognitionScenario.Dictation, "dictation");
+            speechRecognizer.Constraints.Add(dictationConstraint);
+            SpeechRecognitionCompilationResult result = await speechRecognizer.CompileConstraintsAsync();
+            if (result.Status != SpeechRecognitionResultStatus.Success)
+            {
+                checkError.Visibility = Visibility.Visible;
+                errorCheck.Visibility = Visibility.Visible;
+                errorCheck.Text = "Recognition Failed!";
+            }
+
+            // Handle continuous recognition events. Completed fires when various error states occur. ResultGenerated fires when
+            // some recognized phrases occur, or the garbage rule is hit. HypothesisGenerated fires during recognition, and
+            // allows us to provide incremental feedback based on what the user's currently saying.
+            speechRecognizer.ContinuousRecognitionSession.Completed += ContinuousRecognitionSession_Completed;
+            speechRecognizer.ContinuousRecognitionSession.ResultGenerated += ContinuousRecognitionSession_ResultGenerated;
+            speechRecognizer.HypothesisGenerated += SpeechRecognizer_HypothesisGenerated;
+
+        }
+        public async Task Initializesynthesizer()
+        {
+
+           synthesizer = new SpeechSynthesizer();
+
+            isListening = false;
+            dispatcher = this.Dispatcher;
+
+            // select the language display
+            var voices = SpeechSynthesizer.AllVoices;
+            foreach (VoiceInformation voice in voices)
+            {
+                if (voice.Language.Contains(synLangStr))
+                {
+                    synthesizer.Voice = voice;
+                    break;
+                }
+            }
+
+            // Check Microphone Plugged in
+            bool permissionGained = await AudioCapturePermissions.RequestMicrophonePermission();
+            if (!permissionGained)
+            {
+                this.dictationTextBox.Text = "Permission to access capture resources was not given by the user, reset the application setting in Settings->Privacy->Microphone.";
```

Comment:
this comment is not applicable to IoT Core ... there is no settings.  there is no idea of consent on IoT Core.